### DISCUSSION TITLE
Connection lost during request

### DISCUSSION THREAD

**Discussion by hkusulja** [wlcrs/huawei_solar#173] [Setting up the integration]
Hello,

I have new Huawei SUN2000-6KTL-1 , I have used Huawei FusionSolar android app to setup it.
The integrated WiFi is connected to my house WiFi. Firmware is updated (software version V200R001C00SPC124 and MBUS firmware is to V100R001C00SPC320). Modbus tcp is enabled unrestricted. Installater password is set.
I can open TCP connection from computer to port 6607. Slave ID is 1. Additional Huawei power sensor is connected and visible in the app.

After HACS and wlcrs/huawei_solar installation. I am unable to connect - Reading from the inverter failed.

In logs I see:

> logger: custom_components.huawei_solar.config_flow
> Source: custom_components/huawei_solar/config_flow.py:114
> Integration: huawei_solar
> 
> Could not read register value, has another device interrupted the connection?
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 356, in _do_read
>     response = await self._client.protocol.read_holding_registers(
>   File "/usr/local/lib/python3.10/site-packages/pymodbus/client/asynchronous/async_io/__init__.py", line 35, in execute
>     resp = await asyncio.wait_for(req, timeout=self._timeout)
>   File "/usr/local/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
>     return fut.result()
> pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] Connection lost during request
> 

What other devices / connections is possible to have / how to disable them?
Should I disable Huawei cloud connection inside FusionSolar app (“Management System” connection to EU / global servers)?
I can use the mobile app and device commissioning as installer with password on the same/home network. If this “device commissioning” needs to be disabled, how to do it?

**Comment by wlcrs** [comment]
You do not need to disable the connection to the Huawei cloud servers for this integration to work.

Make sure that the integration is the only thing connecting to the inverter. This error only occurs when the modbus connection is stolen from the integration. I have nothing else to add/suggest tbh.

**Comment by hkusulja** [comment]
@wlcrs thank you, this is a new device / inverter in existing network, except android app for configuration nothing was ever connected. Does somebody have more experience with the mentioned inverter and what else could be? maybe some additional setting? newer firmware issue/limitation, or something else.

**Comment by xm4rcell0x** [comment]
have you already tried on port 502 ?

**Comment by hkusulja** [comment]
> have you already tried on port 502 ?

this port was never opened on my device, and is still not.

I tried for days, and seems something inside device is interrupting communication so i can not get modbus tcp to get working (still the same / original error as described above). Maybe something from Huawei, but i do not know what.

**Comment by xm4rcell0x** [comment]
For me the exact opposite...port 502 open and 6607 closed.

**Reply by hkusulja** [reply]
> For me the exact opposite...port 502 open and 6607 closed.

i believe this is up to device type and firmware version, my firmware is latest for this device.

**Reply by galonias** [reply]
> For me the exact opposite...port 502 open and 6607 closed.

Same goes for my Inverter that we recently installed. Only ports opened on the device is 502 and 6606, but the integration are unable to connect, via WLAN, FE or bridged via SUN-TAXXXXX WLAN.

- Dongle,firmware V100R001C00SPC125,
- Inverter, firmware V100R001C00SPC147
- MBUS, firmware V100R001C00SPC323
Logger: huawei_solar.huawei_solar
Source: /usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py:163
First occurred: 08:24:19 (6 occurrences)
Last logged: 12:12:27

Aborting client creation due to error.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 356, in _do_read
    response = await self._client.protocol.read_holding_registers(
  File "/usr/local/lib/python3.10/site-packages/pymodbus/client/asynchronous/async_io/__init__.py", line 35, in execute
    resp = await asyncio.wait_for(req, timeout=self._timeout)
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
    return fut.result()
pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] Connection lost during request

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 157, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 120, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 254, in get
    return (await self.get_multiple([name], slave))[0]
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 296, in get_multiple
    response = await self._read_registers(
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 389, in _read_registers
    result = await _do_read()
  File "/usr/local/lib/python3.10/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 385, in _do_read
    raise ReadException(message) from err
huawei_solar.exceptions.ReadException: Could not read register value, has another device interrupted the connection?

**Comment by hkusulja** [comment]
It seems that Huawei did some changes:
* Starting with firmware SPC125 for devices SUN2000-6KTL-1 , internal WiFi does not support MODBUS anymore officially, and you need to purchase additional Dongle (with ethernet/wifi) where MODBUS should work.

I confirm that I have purchased additional dongle, and dongle is answering on 502 TCP port and not 6607, and I confirm MODBUS is working and this bug/discussion is over.

It is very strange to offer firmware update and limit functionalities.

Those who have M models, already got the dongle with inverter and modbus should work. But those with monophase / L models, need to purchase additional dongle and cannot work with integrated WiFi (with modbus enabled, and over 6607 port which is open), heh...
